clang/lib/StaticAnalyzer/Core/ExprEngineC.cpp

```diff
@@ -264,7 +264,8 @@ ProgramStateRef ExprEngine::handleLValueBitCast(
   }
   // Delegate to SValBuilder to process.
   SVal OrigV = state->getSVal(Ex, LCtx);
-  SVal V = svalBuilder.evalCast(OrigV, T, ExTy);
+  SVal SimplifiedOrigV = svalBuilder.simplifySVal(state, OrigV);
+  SVal V = svalBuilder.evalCast(SimplifiedOrigV, T, ExTy);
```

Comment:
Did you try to simplify after the cast?
To me, it would make more sense to bind a simplified value.
However, I also know that casts are weird at best :D

Reply:
Bound value might not be used at all, so I think lazy simplification would be better.